### PR TITLE
fix: create facts file in vm to support multi-arch builds

### LINKED ISSUE #95: create a `custom.facts` file for multi-arch builds
We need to create `/etc/rhsm/facts/custom.facts` with following content to support multi-arch builds for aarch64 and amd64:
```JSON
{
  "supported_architectures": "aarch64,x86_64"
}
```

### PR DESCRIPTION
**Pull request by dgolovin**
Fix #95.

### CODE REVIEW

**Review by benoitf — APPROVED** [review]
missing telemetry part ?

**Review by vrothberg — COMMENTED** [review]
The brackets "{...}" seem to be missing:

Screenshot 2024-04-12 at 09 39 58

**Comment by vrothberg** [comment]
Just tested.  The brackets are needed:
> Error: Failed to download metadata for repo 'rhel-9-for-x86_64-appstream-beta-rpms': Cannot download repomd.xml: Curl error (77): Problem with the SSL CA cert (path? access rights?) for https://cdn.redhat.com/content/beta/rhel9/9/x86_64/appstream/os/repodata/repomd.xml [error setting certificate file: %(ca_cert_dir)sredhat-uep.pem]
Error:building at STEP "RUN dnf install -y httpd": while running runtime: exit status 1

Try building the following Dockerfile:
```Dockerfile
FROM registry.redhat.io/rhel9-beta/rhel-bootc:9.4

RUN dnf repolist
RUN dnf install -y httpd
```

Run it on an ARM Mac and build the image for AMD64/x86_64.

File: src/podman-cli.ts [hunk 1]
```diff
@@ -31,6 +31,11 @@ const PODMAN_COMMANDS = {
   SM_DEACTIVATE_SUBS: () => `machine ssh sudo subscription-manager unregister`.split(' '),
   MACHINE_STOP: () => 'machine stop'.split(' '),
   MACHINE_START: () => 'machine start'.split(' '),
+  WRITE_FACTS_FILE: () => [
+    'machine',
+    'ssh',
+    'mkdir -p /etc/rhsm/facts/ && echo {\\"supported_architectures\\":\\"aarch64,x86_64\\"} | sudo tee /etc/rhsm/facts/podman-desktop-redhat-account-ext.facts',
```
Comment (benoitf) [inline src/podman-cli.ts:37]: better to use JSON.stringify from a JSON object ?

**Comment by vrothberg** [comment]
The following diff did the trick for me.  Had to create a new podman machine for it work though.  Please test before merging to make sure we verified it on as many machines as possible.

```diff
diff --git a/src/extension.ts b/src/extension.ts
index 7879fd890ed0..3160db3bd9ec 100644
--- a/src/extension.ts
+++ b/src/extension.ts
@@ -347,8 +347,8 @@ export async function activate(context: extensionApi.ExtensionContext): Promise<
               await restartPodmanVM();
             }
             if (!(await isPodmanVmSubscriptionActivated())) {
-              await createOrReuseActivationKey();
               await runWriteFactsFile();
+              await createOrReuseActivationKey();
             }
           }
         },
diff --git a/src/podman-cli.ts b/src/podman-cli.ts
index 3422099367be..bfcf8bd102da 100644
--- a/src/podman-cli.ts
+++ b/src/podman-cli.ts
@@ -34,7 +34,7 @@ const PODMAN_COMMANDS = {
   WRITE_FACTS_FILE: () => [
     'machine',
     'ssh',
-    'mkdir -p /etc/rhsm/facts/ && echo {\\"supported_architectures\\":\\"aarch64,x86_64\\"} | sudo tee /etc/rhsm/facts/podman-desktop-redhat-account-ext.facts',
+    'mkdir -p /etc/rhsm/facts/ && echo \'{"supported_architectures":"aarch64,x86_64"}\' | sudo tee /etc/rhsm/facts/podman-desktop-redhat-account-ext.facts',
   ],
 };
```

**Comment by vrothberg** [comment]
@odockal FYI

**Comment by dgolovin** [comment]
> Just tested. The brackets are needed:

Thanks for testing. Was doing it under windows, there were so many weird errors, but that one worked for me. I have not tested linux and macOS. They are next.

**Comment by dgolovin** [comment]
`\\"` definitely not required, they are just left behind form previous tests.

**Comment by dgolovin** [comment]
> missing telemetry part ?

@benoitf  Telemetry is worked on in different issue.

**Review by benoitf — APPROVED** [review]

**Pull request merged**